client/viewer/js/main.js

```diff
@@ -99,6 +107,19 @@ window.screensInit = function screensInit() {
 	viewer.on('not-connected', () => {
 		DOM.container.classList.add('state-disconnected');
 	});
+
+	viewer.on('ready', function(e){
+		setInterval(function () {
+			updateTitle();
+			updateIDs();
+			DOM.container.classList.toggle('state-disconnected', !viewer.ready());
+			DOM.container.classList.remove('state-active', 'state-hello', 'state-loading');
+			DOM.container.classList.add(viewer.getUrl() ? 'state-active' : (viewer.ready() ? 'state-hello' : 'state-loading'));
```

Comment:
Yeah, sure. I didn't write this bit, I just moved it, but I largely dislike ternary statements.